src/pages/SearchPage.tsx

```diff
@@ -0,0 +1,90 @@
+import { Component, ContextType } from 'react';
+import { BookSearchResponse } from '../api/models';
+import { safeFetch } from '../api/safe-fetch';
+import { OPEN_LIBRARY_URL } from '../api/urls';
+import { ErrorContext } from '../components/ErrorBoundary';
+import { ErrorButton } from '../components/ErrorButton';
+import { Results } from '../components/Results';
+import { TopControls } from '../components/TopControls';
+
+export interface SearchPageState {
+  response?: BookSearchResponse;
+  isLoading: boolean;
+}
+
+const PAGE_SIZE = 5;
+
+export class SearchPage extends Component<unknown, SearchPageState> {
+  state: SearchPageState = {
+    isLoading: false,
+  };
+
+  static contextType = ErrorContext;
+  declare context: ContextType<typeof ErrorContext>;
+  prevContextValue?: typeof this.context;
+
+  componentDidUpdate() {
+    if (this.context.error !== this.prevContextValue?.error) {
+      if (this.context.error) {
+        this.setState({ isLoading: false, response: undefined });
+      }
+      this.prevContextValue = this.context;
+    }
+  }
+
+  handleSearch = async (search: string) => {
+    if (!search) return;
+
+    this.context.setError(null);
+    this.setState({ ...this.state, isLoading: true });
```

Comment:
```suggestion
    this.setState({ isLoading: true });
```
react merges state values with current/previous one. So it's enough to pass only properties which need to be updated

Reply:
thanks, good to know!

[refactor: refactor SearchPage](https://github.com/programister42/rs-react-app/pull/1/commits/3bbf8db772e0acf147296bad6a0868ab375a9eb6)